maven-plugins/stager-maven-plugin/src/main/java/io/helidon/build/maven/stager/Container.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.build.maven.stager;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * {@link StagingAction} Container and execute them with provided executor.
+ * @param <T> StagingAction
+ */
+public class Container<T extends StagingAction> implements StagingAction {
+
+    private static final Collection<Callable<CompletionStage<Void>>> TASKS_QUEUE = new ConcurrentLinkedQueue<>();
```

Comment:
`Container.executor`, `Container.submit` shouldn't be static ; that feels wrong.

We should move the concurrent related stuff in a different class, and pass the instance around.
How about adding that to the `StagingContext` seems like it is a good fit.